### ISSUE TITLE
setup-conda action failing for Ubuntu tests

### ISSUE THREAD

**Issue by dcdenu4** [natcap/pygeoprocessing#44]
When working on Issue #16, migrating to GDAL 3 seems to cause the Ubuntu tests to fail using `setup-conda`. It appears that `setup-conda` does not properly set up that environment for **GDAL_DATA** and **PROJ_LIB** environment variables, which then give the following test errors:

```
ERROR 1: PROJ: proj_create_from_database: Open of /usr/share/miniconda/share/proj failed
ERROR 1: PROJ: proj_create_from_database: Open of /usr/share/miniconda/share/proj failed
```

It appears that even with GDAL 2 those environment variables were not being set properly, but the tests did not fail. One theory for this, is that before GDAL 3 EPSG codes were stored in a `csv` file in `gdal/data/` somewhere, opposed to GDAL 3 that now uses sqlite DBs and let's PROJ handle the codes.

**Comment by dcdenu4** [comment]
An alternative to `setup-conda` is[setup-miniconda](https://github.com/goanpeca/setup-miniconda) which I prototyped and succesfully ran Ubuntu tests. I could confirm that the environment variables **GDAL_DATA** and **PROJ_LIB** were being set. However, `windows-latest` tests fail under this action method. 

See the issue here: https://github.com/goanpeca/setup-miniconda/issues/15

**Comment by dcdenu4** [comment]
It looks like a minor change and recommendation from the `setup-miniconda` maintainer has the Windows build and tests working. The only caveat for now is that we'll need to specify a `pythonpackage-condarc.yml` file somewhere. In this file `channels` is defined to include `conda-forge` and is read by `pythonpackage.yml`. 
@phargogh @richpsharp recommendations on where this should live in the repo tree? I currently have it living in `.github/`.

I ultimately think this Actions repo is preferred over our current use of `setup-conda`.

The maintainer is working on including `channels` and additional conda specifications directly into the workflow file but it is not yet implemented.

**Comment by dcdenu4** [comment]
Actually, might be worth just sitting on this for awhile as hopefully we won't need to add that config file. 

https://github.com/goanpeca/setup-miniconda/issues/4#issuecomment-598729030

**Comment by phargogh** [comment]
@dcdenu4 this looks really interesting!  Is this the sort of file that is useful outside of a build context?  Like, should we be using such a condarc file to help with creating our own conda environments?

If so, it might be worth keeping and using the file, and the repo root could be a good home for it.  If it's specific to github actions, I'd suggest `.github/workflows` since that's where basically all of the CI stuff is at the moment.

**Comment by dcdenu4** [comment]
Fixed in #49
